tests/sympc/store/crypto_primitive_provider_test.py

```diff
@@ -174,57 +170,203 @@ def test_primitive_logging_model(get_clients) -> None:
-        }
-    else:
-        g_kwargs = {"n_values": 4}
-        p_kwargs = {}
+    p_kwargs = {"a_shape": (2, 10), "b_shape": (2, 10)}
+    g_kwargs = {"a_shape": (2, 10), "b_shape": (2, 10), "nr_parties": 2}
 
     CryptoPrimitiveProvider.start_logging()
     CryptoPrimitiveProvider.generate_primitives(
         sessions=session.session_ptrs,
-        op_str=ops[0],
+        op_str="beaver_mul",
         p_kwargs=p_kwargs,
         g_kwargs=g_kwargs,
     )
     primitive_log = CryptoPrimitiveProvider.stop_logging()
-    expected_log = json.dumps({ops[0]: [ops[1]]})
+    expected_log = {"beaver_mul": [(p_kwargs, g_kwargs)]}
 
     assert expected_log == primitive_log
+
+
+def test_primitive_logging_beaver_matmul(get_clients) -> None:
+    clients = get_clients(2)
+    session = Session(parties=clients)
+    SessionManager.setup_mpc(session)
+
+    p_kwargs = {"a_shape": (2, 3), "b_shape": (3, 10)}
+    g_kwargs = {"a_shape": (2, 3), "b_shape": (3, 10), "nr_parties": 2}
+
+    CryptoPrimitiveProvider.start_logging()
+    CryptoPrimitiveProvider.generate_primitives(
+        sessions=session.session_ptrs,
+        op_str="beaver_matmul",
+        p_kwargs=p_kwargs,
+        g_kwargs=g_kwargs,
+    )
+    primitive_log = CryptoPrimitiveProvider.stop_logging()
+    expected_log = {"beaver_matmul": [(p_kwargs, g_kwargs)]}
+
+    assert expected_log == primitive_log
+
+
+def test_primitive_logging_beaver_conv2d(get_clients) -> None:
+    clients = get_clients(2)
+    session = Session(parties=clients)
+    SessionManager.setup_mpc(session)
+
+    p_kwargs = {"a_shape": (1, 1, 28, 28), "b_shape": (5, 1, 5, 5)}
+    g_kwargs = {"a_shape": (1, 1, 28, 28), "b_shape": (5, 1, 5, 5), "nr_parties": 2}
+
+    CryptoPrimitiveProvider.start_logging()
+    CryptoPrimitiveProvider.generate_primitives(
+        sessions=session.session_ptrs,
+        op_str="beaver_conv2d",
+        p_kwargs=p_kwargs,
+        g_kwargs=g_kwargs,
+    )
+    primitive_log = CryptoPrimitiveProvider.stop_logging()
+    expected_log = {"beaver_conv2d": [(p_kwargs, g_kwargs)]}
+
+    assert expected_log == primitive_log
+
+
+def test_generate_primitive_from_dict_beaver_matmul(get_clients) -> None:
+    clients = get_clients(2)
+    session = Session(parties=clients)
+    SessionManager.setup_mpc(session)
+
+    primitive_log = {
+        "beaver_matmul": [
+            (
+                {"a_shape": (2, 3), "b_shape": (3, 10)},
+                {"a_shape": (2, 3), "b_shape": (3, 10), "nr_parties": 2},
+            )
+        ]
+    }
+
+    CryptoPrimitiveProvider.generate_primitive_from_dict(
+        primitive_log=primitive_log, session=session
+    )
+
+    a_shape = (2, 3)
+    b_shape = (3, 10)
+
+    key = f"beaver_matmul_{a_shape}_{b_shape}"
+
+    store_client_1 = session.session_ptrs[0].crypto_store.store.get()
+    store_client_2 = session.session_ptrs[1].crypto_store.store.get()
+
+    shape_client_1 = tuple(store_client_1.get(key)[0][0].shape)
```

Comment:
MAYBE: You also need to check the other shape, right?

Reply:
other shape ? I did not get 😟

Reply:
done